### PR TITLE
FileManager: Add links in file properties to open links in their real directory

### PR DESCRIPTION
**Pull request by lxea**
This adds a button if a file is a link to open a file manager in the original  files directory
20201221_14h05m18s_grim

### CODE REVIEW

**Comment by diversys** [comment]
Maybe Location and Link target path sections can be linkified?

**Comment by AlexKraak** [comment]
> Maybe Location and Link target path sections can be linkified?

I'll give this a try, that seems like itd be better

File: Libraries/LibGUI/Link.cpp [hunk 1]
```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright (c) 2018-2020, Andreas Kling <kling@serenityos.org>
```
Comment (linusg) [inline Libraries/LibGUI/Link.cpp:2]: Andreas didn't write this, you did :)

File: Libraries/LibGUI/Link.h [hunk 2]
```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright (c) 2018-2020, Andreas Kling <kling@serenityos.org>
```
Comment (linusg) [inline Libraries/LibGUI/Link.h:2]: Same as above

File: Libraries/LibGUI/Link.cpp [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+    if (on_click) {
+        on_click();
+    }
+}
+
+void Link::paint_event(PaintEvent& event)
+{
+    Label::paint_event(event);
+    GUI::Painter painter(*this);
+    painter.draw_line({ 0, rect().bottom() }, { font().width(text()), rect().bottom() }, Color::MidGray);
```
Comment (linusg) [inline Libraries/LibGUI/Link.cpp:47]: this is wrong as it won't work properly with themes other than the default one, please use `Gfx::Palette` (accessible via `Widget::palette()`)

**Review by linusg — CHANGES_REQUESTED** [review]

File: Libraries/LibGUI/Link.cpp (hunk 3 above)
Comment (AlexKraak) [reply Libraries/LibGUI/Link.cpp]: Ah thank you, I wasnt sure what to look for for this!

**Comment by diversys** [comment]
Looks nice! Just a few more thoughts:
How about making links coloured by default and underline them on mouse over? 
Is there a tooltip if a link is longer than window width?
There is something wrong with separators below Browser text field and above permission view.

**Comment by diversys** [comment]
Also is scroll to selection after clicking links implemented?

**Comment by lxea** [comment]
> How about making links coloured by default and underline them on mouse over?

20201224_13h11m06s_grim
I have this now, if you hover over it it shows the underline and the tooltip, 

> Is there a tooltip if a link is longer than window width?

im not sure how to make it so the tool tip only shows if its too long but ill give it another stab in a bit

> There is something wrong with separators below Browser text field and above permission view.

Not sure what you mean here, i dont think they should have changed

> Also is scroll to selection after clicking links implemented?

I dont think this exists yet,

**Comment by diversys** [comment]
This looks nifty! :)

**Comment by lxea** [comment]
Latest commit, it looks the same as last screenshot but itll only show the tooltip if the text is wider than the window

**Pull request merged**